libs/feature/search/src/lib/utils/links/link-utils.ts

```diff
@@ -70,3 +130,13 @@ export function checkFileFormat(
     ('url' in link && new RegExp(`[./]${format}`, 'i').test(link.url))
   )
 }
+
+export function getBadgeColor(linkFormat: string): string | void {
+  for (const format in FORMATS) {
+    for (const alias of FORMATS[format].extensions) {
+      if (new RegExp(`${alias}`, 'i').test(linkFormat))
+        return FORMATS[format].color
+    }
+  }
+  return 'var(--color-primary)' // Default color ?
```

Comment:
it might be better to keep this default value in the CSS, and make sure that specific colors will have precedence over it